libtsuba/src/GlobalState.cpp

```diff
@@ -123,16 +123,22 @@ tsuba::OneHostOnly(const std::function<katana::Result<void>()>& cb) {
   // execution of OneHostOnly
   Comm()->Barrier();
 
+  std::optional<katana::Result<void>> res;
+
   bool failed = false;
   if (Comm()->ID == 0) {
-    auto res = cb();
+    res = cb();
     if (!res) {
-      KATANA_LOG_ERROR("OneHostOnly operation failed: {}", res.error());
       failed = true;
     }
   }
 
   if (Comm()->Broadcast(0, failed)) {
+    if (!res) {
+      return KATANA_ERROR(
```

Comment:
Should this be: `res.WithContext(ErrorCode::MpiError, "failure in single host execution")`

We have reclassifying `WithContext` and this seems like the exact case to use it.